addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/messages/RFXComUndecodedRFMessage.java

```diff
@@ -0,0 +1,198 @@
+
+    public SubType subType = SubType.UNKNOWN;
+    public byte[] rawPayload = new byte[0];
+
+    public RFXComUndecodedRFMessage() {
+        packetType = PacketType.UNDECODED_RF_MESSAGE;
+    }
+
+    public RFXComUndecodedRFMessage(byte[] message) {
+        encodeMessage(message);
+    }
+
+    @Override
+    public String toString() {
+
+        String str = "";
+
+        str += super.toString();
+        str += "\n - Sub type = " + subType;
+        str += "\n - Id = " + getDeviceId();
+        str += "\n - Message = " + DatatypeConverter.printHexBinary(rawMessage);
+
+        return str;
+    }
+
+    @Override
+    public void encodeMessage(byte[] message) {
+
+        super.encodeMessage(message);
+
+        subType = SubType.fromByte(super.subType);
+        rawPayload = Arrays.copyOfRange(rawMessage, 4, rawMessage.length);
+    }
+
+    @Override
+    public byte[] decodeMessage() {
+
+        final int rawPayloadLen = Math.min(rawPayload.length, 33);
+        byte[] data = new byte[4 + rawPayloadLen];
+
+        data[0] = (byte) (data.length - 1);
+        data[1] = RFXComBaseMessage.PacketType.UNDECODED_RF_MESSAGE.toByte();
+        data[2] = subType.toByte();
+        data[3] = seqNbr;
+
+        for (int i = 0; i < rawPayloadLen; i++) {
+            data[i + 4] = rawPayload[i];
+        }
+        return data;
+    }
+
+    @Override
+    public String getDeviceId() {
+        return "UNDECODED";
+    }
+
+    @Override
+    public State convertToState(RFXComValueSelector valueSelector) throws RFXComException {
+
+        if (valueSelector.getItemClass() == StringItem.class) {
+            if (valueSelector == RFXComValueSelector.RAW_MESSAGE) {
+                return new StringType(DatatypeConverter.printHexBinary(rawMessage));
+            } else if (valueSelector == RFXComValueSelector.RAW_PAYLOAD) {
+                return new StringType(DatatypeConverter.printHexBinary(rawPayload));
+            }
+        }
+
+        throw new RFXComException("Can't convert " + valueSelector + " to " + valueSelector.getItemClass());
+    }
+
+    @Override
+    public void setSubType(Object subType) throws RFXComException {
+        throw new RFXComException("Not supported");
+    }
+
+    @Override
+    public void setDeviceId(String deviceId) throws RFXComException {
+        throw new RFXComException("Not supported");
+    }
+
+    @Override
+    public void convertFromState(RFXComValueSelector valueSelector, Type type) throws RFXComException {
+        throw new RFXComException("Not supported");
+    }
+
+    @Override
+    public Object convertSubType(String subType) throws RFXComException {
+
+        for (SubType s : SubType.values()) {
+            if (s.toString().equals(subType)) {
+                return s;
+            }
+        }
+
+        // try to find sub type by number
+        try {
+            return SubType.values()[Integer.parseInt(subType)];
```

Comment:
O oh you use the array index here instead of the byte, for now the array indices match the bytes but no long term guarantees, use the static method please.